freight/checks/github_apps.py

```diff
@@ -41,19 +41,38 @@ def check(self, app, sha, config):
             "Authorization": f"token {token}",
         }
 
-        resp = http.get(url, headers=headers)
+        params = {
+            "per_page": 100,
+            "page": 1,
+        }
+
+        total_count = None
+        check_runs = []
+
+        while True:
```

Comment:
When does the loop break other than with a `CheckFailed` exception?

Reply:
Here ~ https://github.com/getsentry/freight/pull/251/files#diff-d152e8fa3194e627549966413aa22ebc073cc3ba9f030ec97c976d8084eeb659R69-R72

when per_page x page >= total count